GA/post.py

```
@@ -0,0 +1,734 @@
+
+        # For multi-objective probems, we plot each objective separately.
+        if avg_objectives.ndim > 1 and avg_objectives.shape[1] >1:
+            n_obj = avg_objectives.shape[1]
+            for i in range(n_obj):
+                plt.plot(n_evals, generational_optimum[i], "-x", label=f'Generational optimum for objective {i}')
+                plt.errorbar(n_evals, avg_objectives[:,i], yerr=3*std_objectives[:,i], fmt="-*", label=f"Generational average for objective {i}", capsize=4, capthick=1.5)      
+        else:
+            avg_objectives = avg_objectives.squeeze()
+            std_objectives = std_objectives.squeeze()
+            plt.plot(n_evals, generational_optimum, "-x", label='Generational optimum')
+            plt.errorbar(n_evals, avg_objectives, yerr=std_objectives, fmt="-*", label=f"Generational average", capsize=4, capthick=1.5)
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)